packages/hash/frontend/src/components/Button.tsx

```diff
@@ -1,100 +1,101 @@
+// eslint-disable-next-line no-restricted-imports
+import Link from "next/link";
 import {
   Box,
   // eslint-disable-next-line no-restricted-imports
   Button as MuiButton,
   ButtonProps as MuiButtonProps,
 } from "@mui/material";
-import { FC, forwardRef } from "react";
+import { VFC, FC, forwardRef, useMemo } from "react";
+import { LoadingSpinnerIcon } from "./icons";
+import { isHrefExternal } from "./Link";
 
-// @todo-mui update this to match the loading state in the design
-// inspired by https://github.com/loadingio/css-spinner/blob/master/src/ellipsis/index.styl
-const loadingAnimation = (
-  <Box
-    sx={{
-      display: "inline-block",
-      position: "relative",
-      width: 80,
-      height: 13,
-      "> div": {
-        position: "absolute",
-        top: 0,
-        width: 12,
-        height: 12,
-        borderRadius: "50%",
-        background: "currentColor",
-        animationTimingFunction: "cubic-bezier(0, 1, 1, 0)",
-        "&:nth-child(1)": {
-          left: 8,
-          animation: "lds-ellipsis1 0.6s infinite",
-        },
-        "&:nth-child(2)": {
-          left: 8,
-          animation: "lds-ellipsis2 0.6s infinite",
-        },
-        "&:nth-child(3)": {
-          left: 32,
-          animation: "lds-ellipsis2 0.6s infinite",
-        },
-        "&:nth-child(4)": {
-          left: 56,
-          animation: "lds-ellipsis3 0.6s infinite",
-        },
-      },
-      "@keyframes lds-ellipsis1": {
-        "0%": {
-          transform: "scale(0)",
-        },
-        "100%": {
-          transform: "scale(1)",
-        },
-      },
-      "@keyframes lds-ellipsis3": {
-        "0%": {
-          transform: "scale(1)",
-        },
-        "100%": {
-          transform: "scale(0)",
-        },
-      },
-      "@keyframes lds-ellipsis2": {
-        "0%": {
-          transform: "translate(0, 0)",
-        },
-        "100%": {
-          transform: "translate(24px, 0)",
-        },
-      },
-    }}
-  >
-    <div />
-    <div />
-    <div />
-    <div />
-  </Box>
-);
+const LoadingContent: VFC<{
+  withText: boolean;
+  isXS: boolean;
+}> = ({ withText, isXS }) => {
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        alignItems: "center",
+      }}
+    >
+      <LoadingSpinnerIcon
+        sx={{
+          width: 16,
+          height: 16,
+          animation: "spin 1.1s linear infinite",
+          "@keyframes spin": {
+            "0%": {
+              transform: "rotate(0)",
+            },
+            "100%": {
+              transform: "rotate(360deg)",
+            },
+          },
+        }}
+      />
+
+      {withText && (
+        <Box
```

Comment:
This is because I want the `span` to inherit the font styles set on the button depending on the variant/size applied. For example large buttons use `largeTextLabels` but small/medium buttons use `smallTextLabels`. Using typography here means I have to re-account for things like the button size/variants here when setting the typography variant. We shouldn't need to do that since all these are handled in the [button's styling](https://github.com/hashintel/hash/blob/0ac9499191407b872b00265e2c8e93f8f24c87dd/packages/hash/frontend/src/theme/components/inputs/MuiButtonThemeOptions.ts#L159-L170)

Reply:
Arguably you wouldn't have had to insert the typography styling into the button styling if you would've used a `Typography` field here, and it would've resulted in a more consistent re-use of our typography styling, and it would mean MUI doesn't have to duplicate the styling in two places when generating CSS classes.

Reply:
I agree, but the typography is tied to/dependent on the size of the button. I think it's better to have all size styles in one place (the button styling) than divide in two places (button styling, and Typography component's variant). It is pretty common in normal css to target and attach the font's styling to the containing button and not the child text